gammapy/estimators/energydependence.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+from gammapy.datasets import Datasets
+from gammapy.modeling import Fit
+from gammapy.modeling.models import FoVBackgroundModel
+from gammapy.modeling.selection import TestStatisticNested
+from gammapy.stats.utils import ts_to_sigma
+from .core import Estimator
+
+
+def weighted_chi2_parameter_results(table_edep, parameter="sigma"):
+    """Calculate the weighted chi2 value for the parameter of interest"""
+
+    values = table_edep[f"{parameter}"][1:]
+    errors = table_edep[f"{parameter}_err"][1:]
+
+    weights = 1 / errors**2
+    avg = np.average(values, weights=weights)
+
+    chi2_value = np.sum((values - avg) ** 2 / errors**2)
+    df = len(values) - 1
+    sigma_value = ts_to_sigma(chi2_value, df)
+
+    chi2_result = {}
+    chi2_result[f"chi2 {parameter}"] = [chi2_value]
+    chi2_result["df"] = [df]
+    chi2_result["significance"] = [sigma_value]
+
+    return chi2_result
+
+
+class EnergyDependenceEstimator(Estimator):
+    """Test if there is any energy-dependent morphology in a map dataset for a given set of energy bins.
+
+    Parameters
+    ----------
+    energy_edges : `~astropy.units.Quantity`
+        Energy edges for the energy-dependence test.
+    source : str or int
+        For which source in the model to compute the estimator.
+    fit : `Fit`
+        Fit instance specifying the backend and fit options.
```

Comment:
In the previous PR, we were speaking about adding in the docstring the list of available fitting backend for all estimators. Can you put an issue to remind that?

Reply:
I don't think this is a good idea to duplicate the list of available backends in the documentation. This will become impossible to maintain.
What should be done is to put a proper link to `~gammapy.modeling.Fit` in the argument that will directly link to the list of available backends.